lib/cocoapods/generate/podfile_generator.rb

```diff
@@ -189,6 +189,20 @@ def use_modular_headers?
         end
       end
 
+      # @return [Boolean, Hash]
+      #         the value to use for `use_frameworks!` DSL directive
+      #
+      def use_frameworks_value
+        return configuration.use_frameworks? unless configuration.use_podfile?
+        target_definition_list.map do |target_definition|
+          if target_definition.respond_to?(:build_type) # CocoaPods > 1.9
+            { :linkage => target_definition.build_type == BuildType.dynamic_framework ? :dynamic : :static}
+          else
+            true
```

Comment:
If we are reading the `Podfile` then this should be reading the `target_definition.uses_frameworks?` flag but the tests were all "wrong" in that sense since the code _always_ was just reading the value from the configuration instead, i.e passing `--use-libraries`.

There was never support of setting `use_frameworks!` based off the `Podfile`.

Reply:
Going to see if I can update to do the right thing instead.